File: mall-sellergoods-interface/src/main/java/com/company/mall/sellergoods/service/BrandService.java
package com.company.mall.sellergoods.service;

import com.company.entity.PageResult;
import com.company.mall.pojo.TbBrand;

import java.util.List;

/**
 * @description: 品牌管理接口类
 * @author: chunguang.yao
 * @date: 2019-06-30 12:51
 */
public interface BrandService {

    /**
     * 查询所有品牌
     * @return
     */
    List<TbBrand> findAll();

    /**
     * 返回分页列表
     * @param pageNum 当前页
     * @param pageSize 每页的记录数
     * @return
     */
    PageResult findPage(int pageNum, int pageSize);

    /**
     * 增加品牌
     * @param brand 品牌对象
     */
    void addBrand(TbBrand brand);

    /**
     * 根据id查询品牌
     * @param id
     * @return
     */
    TbBrand findBrandById(long id);

    /**
     * 更新品牌信息
     * @param brand
     */
    void updateBrand(TbBrand brand);

    /**
     * 根据id删除品牌信息
     * @param ids
     */
    void deleteBrandById(long[] ids);

    /**
     * 带条件的分页查询
     * @param brand
     * @param pageNum
     * @param pageSize
     * @return
     */
    PageResult findPageByCondition(TbBrand brand, int pageNum, int pageSize);
}

File: mall-manager-web/src/main/webapp/js/service/brandService.js
// 品牌服务层
app.service('brandService', function ($http) {
    // 读取所有数据
    this.findAll = function() {
        return $http.get('../brand/findAll.do');
    }

    // 分页查询数据
    this.findPage = function(page, size) {
        return $http.get('../brand/findPage.do?page=' + page + '&rows=' + size);
    }

    // 增加品牌
    this.addBrand = function(entity) {
        return $http.post('../brand/addBrand.do', entity);
    }

    // 更新品牌
    this.updateBrand = function(entity) {
        return $http.post('../brand/updateBrand.do', entity);
    }

    // 根据id查询品牌实体
    this.findBrandById = function(id) {
        return $http.get('../brand/findBrandById.do?id=' + id);
    }

    // 删除品牌信息
    this.deleteBrand = function(ids) {
        return $http.get('../brand/deleteBrandById.do?ids=' + ids);
    }

    // 搜索
    this.search = function(page, rows, searchEntity) {
        return $http.post('../brand/search.do?page=' + page + '&rows=' + rows, searchEntity);
    }

});
